clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/validators/StandardAppEngineAttributeValidator.groovy

```diff
@@ -261,4 +257,54 @@ class StandardAppEngineAttributeValidator {
       return true
     }
   }
+
+  def validateShardBy(AppEngineTrafficSplit split, Boolean migrateTraffic, String attribute) {
+    if (!split) {
+      return true
+    }
+
+    def shardByExists = { validateNotEmpty(split.shardBy, attribute) }
```

Comment:
Oh, I think there used to be more logic in there, so I can just `validateNotEmpty` directly. But I want `validateNotEmpty` to run only in certain cases, since there are times when I don't care if `split.shardBy` is empty, and in those cases I don't want `validateNotEmpty` to add error messages. If that makes sense.